### PR TITLE
Auto-configure inAppIncludes in Spring Boot integration.

### PR DESCRIPTION
**Pull request by maciejwalkowiak**
## :loudspeaker: Type of change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring

## :scroll: Description

Auto-configure `inAppIncludes` in Spring Boot integration by getting a package name from `@SpringBootApplication` or `@SpringBootConfiguration` annotated classes.

## :bulb: Motivation and Context

Less configuration and better experience for our users.

## :pencil: Checklist

- [x] I reviewed submitted code
- [x] I added tests to verify changes
- [x] All tests passing
- [ ] No breaking changes

### CODE REVIEW

**Review by bruno-garcia — APPROVED** [review]
LGTM

File: sentry-spring-boot-starter/src/main/java/io/sentry/spring/boot/InAppIncludesResolver.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+import org.springframework.context.ApplicationContextAware;
+
+/**
+ * Resolves {@link SentryProperties} inAppIncludes by getting a package name from a class annotated
+ * with {@link SpringBootConfiguration} or another annotation meta-annotated with {@link
+ * SpringBootConfiguration} like {@link SpringBootApplication}.
+ */
+@Open
+class InAppIncludesResolver implements ApplicationContextAware {
+  @Nullable private ApplicationContext applicationContext;
```
Comment (marandaneto) [inline sentry-spring-boot-starter/src/main/java/io/sentry/spring/boot/InAppIncludesResolver.java:23]:
```suggestion
  private @Nullable ApplicationContext applicationContext;
```

**Review by marandaneto — APPROVED** [review]
LGTM
quick question: why `No breaking changes` is not checked? it's not really a breaking change, right?

**Comment by maciejwalkowiak** [comment]
It alters the previous behavior so not really breaking but once users upgrade and they didnt set this propery manually before they will see a change. Does it qualify as breaking?

**Comment by marandaneto** [comment]
> It alters the previous behavior so not really breaking but once users upgrade and they didn't set this propery manually before they will see a change. Does it qualify as breaking?

I see, luckily if manually set and now automatically detected, would be still the same values, IMO it's not a breaking change but got your point, maybe let's just describe this in the release notes so we don't need to minor bump, but only patch bump, wdyt/

**Review by bruno-garcia — APPROVED** [review]
:shipit: ?

**Pull request merged**